app/controllers/books_controller.rb

```diff
@@ -0,0 +1,52 @@
+class BooksController < ApplicationController
+  before_action :set_book, only: %i[show update destroy]
+
+  # GET /books
+  def index
+    @books = Book::Reducer.apply(params)
+
+    render json: @books
+  end
+
+  # GET /books/1
+  def show
+    render json: @book
+  end
+
+  # POST /books
+  def create
+    @book = Book.new(book_params)
+
+    if @book.save
+      render json: @book, status: :created, location: @book
+    else
+      render json: @book.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /books/1
+  def update
+    if @book.update(book_params)
+      render json: @book
+    else
+      render json: @book.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /books/1
+  def destroy
+    @book.destroy
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_book
+    @book = Book.find(params[:id])
```

Comment:
We use something like [Memorization pattern](https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/) to do this :ok_man:

Reply:
```rb
def book
  @book ||= Book.find(params.require(:id))
end
```